src/ast/data_type.rs

```diff
@@ -716,7 +724,15 @@ impl fmt::Display for DataType {
             DataType::Unspecified => Ok(()),
             DataType::Trigger => write!(f, "TRIGGER"),
             DataType::AnyType => write!(f, "ANY TYPE"),
-            DataType::Table(fields) => write!(f, "TABLE({})", display_comma_separated(fields)),
+            DataType::Table(fields) => {
+                if fields.is_empty() {
+                    return write!(f, "TABLE");
+                }
+                write!(f, "TABLE({})", display_comma_separated(fields))
```

Comment:
I think instead of using the `is_empty` we can make the `fields` value an `Option` and skip the parenthesis only if `fields` is `None`. Otherwise we could run into issues if empty fields is allowed in the other variant of this datatype